napari/utils/events/containers/_evented_list.py

```diff
@@ -352,3 +410,36 @@ def reverse(self) -> None:
         # it would just emit a "changed" event for each moved index in the list
         self._list.reverse()
         self.events.reordered(value=self)
+
+    def _update_inplace(self, other):
+        # inplace updating only happens from parent after validation, so no need to validate here.
+        with self._no_validation():
+            self[:] = list(other)
+
+    def _validate_with_parent(self, value):
+        if self._parent is not None and self._validate:
+            parent = self._parent[0]
+            field = self._parent[1]
+            pdict = parent.dict()
+            pdict[field] = value
+            new = parent._pre_validate(pdict)
+            # TODO this actually fails if validation causes a field other than `field` to
+            # change; that change won't be upstreamed and we break...
+            return new[field]
+        else:
+            return value
+
+    @contextmanager
+    def _no_validation(self):
+        self._validate = False
+        yield
+        self._validate = True
+
+    def _uneventful(self):
+        ret = list()
```

Comment:
I'm guessing that this new list creation is at least partially responsible for the increase in time we're seeing. I wonder if there is a way to get around that / reuse something private instead. Maybe @andy-sweet can look here

Reply:
I did some performance monitoring (though I'm not good at it), and I don't think this is a big concern. The main culprits seemed to be:
- hooking up events when creating all the evented classes (uses a lof of `inspect.signature`)
- validating models multiple times

though I'd wait for someone else to check on this to make sure I understood the outputs :P